server/channels/store/searchtest/file_info_layer.go

```diff
@@ -1185,6 +1190,67 @@ func testFileInfoSupportWildcards(t *testing.T, th *SearchTestHelper) {
 	})
 }
 
+func testFileInfoAlwaysSupportWildcards(t *testing.T, th *SearchTestHelper) {
+	post, err := th.createPost(th.User.Id, th.ChannelBasic.Id, "testmessage", "", model.PostTypeDefault, 0, false)
+	require.NoError(t, err)
+	defer th.deleteUserPosts(th.User.Id)
+
+	p1, err := th.createFileInfo(th.User.Id, post.Id, "search post", "search post", "jpg", "image/jpeg", 0, 0)
+	require.NoError(t, err)
+	p2, err := th.createFileInfo(th.User.Id, post.Id, "searching", "searching", "jpg", "image/jpeg", 0, 0)
+	require.NoError(t, err)
+	p3, err := th.createFileInfo(th.User.Id, post.Id, "long post", "first text and second text and last text", "jpg", "image/jpeg", 0, 0)
+	require.NoError(t, err)
+	_, err = th.createFileInfo(th.User.Id, post.Id, "another post", "another post", "jpg", "image/jpeg", 0, 0)
+	require.NoError(t, err)
+	_, err = th.createFileInfo(th.User.Id, post.Id, "another long post", "another text and second text and last text", "jpg", "image/jpeg", 0, 0)
+	require.NoError(t, err)
+	defer th.deleteUserFileInfos(th.User.Id)
+
+	t.Run("supports wildcard search without wildcards", func(t *testing.T) {
+		params := &model.SearchParams{
+			Terms: "search",
+		}
+		results, err := th.Store.FileInfo().Search([]*model.SearchParams{params}, th.User.Id, th.Team.Id, 0, 20)
+		require.NoError(t, err)
+
+		require.Len(t, results.FileInfos, 2)
+		th.checkFileInfoInSearchResults(t, p1.Id, results.FileInfos)
+		th.checkFileInfoInSearchResults(t, p2.Id, results.FileInfos)
+	})
+
+	t.Run("support wildcard search with another term placed after without wildcards", func(t *testing.T) {
+		params := &model.SearchParams{
+			Terms: "sear post",
```

Comment:
Behaviourally speaking; I'm unsure whether we should treat this as a `sear* post`. I understand adding wildcard to the end of the search term but this didn't sound right to me.

Reply:
I think it's awkward, too. I removed the test case and fixed lint.